lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -5717,6 +5717,13 @@ bool LLVMToSPIRVBase::transExecutionMode() {
           BF->addExecutionMode(BM->add(
               new SPIRVExecutionMode(BF, static_cast<ExecutionMode>(EMode))));
       } break;
+      case spv::internal::ExecutionModeNamedSubgroupSizeINTEL: {
+        if (!BM->isAllowedToUseExtension(ExtensionID::SPV_INTEL_subgroup_requirements))
+          break;
+        BM->addExtension(ExtensionID::SPV_INTEL_subgroup_requirements);
+        BM->addCapability(spv::internal::CapabilitySubgroupRequirementsINTEL);
```

Comment:
It seem like we have inconsistency in our code base in terms of adding capabilities and extensions. Yet I'd recommend to modify lib/SPIRV/libSPIRV/SPIRVEnum.h to add mapping between execution mode and the capability. Optionally, please consider modifying getRequiredExtension from SPIRVCapability class (lib/SPIRV/libSPIRV/SPIRVEntry.h) to add mapping between the extension and capability, but it's some sort of a grey area, is it seems like we have almost never used it.